The check for node_modules no longer works

## Description of the issue
When you run `npm start` on the `main` branch without having run `npm install` you get errors about `nodemon`, instead of our nice message to users to run `npm install`

## Steps to reproduce the issue
Run `npm start` without having run `npm install`

## Actual vs expected behaviour
Instead of a nice message to run `npm install` (provided by the function `checkFiles`) you get this:

```
> govuk-prototype-kit@12.1.0 dev
> node start

node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'nodemon/lib/utils/colour'
Require stack:
- /Users/joelanman/projects/govuk-prototype-kit/lib/build/tasks.js
- /Users/joelanman/projects/govuk-prototype-kit/start.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/joelanman/projects/govuk-prototype-kit/lib/build/tasks.js:1:16)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/joelanman/projects/govuk-prototype-kit/lib/build/tasks.js',
    '/Users/joelanman/projects/govuk-prototype-kit/start.js'
  ]
}
```

--- a/__tests__/spec/start.js
+++ b/__tests__/spec/start.js
@@ -1,0 +1,28 @@
+/* eslint-env jest */
+
+const child_process = require('child_process') // eslint-disable-line camelcase
+const fs = require('fs')
+const path = require('path')
+
+const utils = require('./utils')
+
+describe('npm run start', () => {
+  const tmpDir = utils.mkdtempSync()
+
+  describe('prestart', () => {
+    it('checks whether node modules are installed', () => {
+      const testDir = path.join(tmpDir, 'check-node-modules-exists')
+      utils.mkPrototypeSync(testDir)
+
+      expect(() => {
+        child_process.execSync('npm run start', { cwd: testDir })
+      }).toThrow('ERROR: Node module folder missing. Try running `npm install`')
+
+      fs.mkdirSync(path.join(testDir, 'node_modules'))
+
+      expect(() => {
+        child_process.execSync('npm run predev', { cwd: testDir })
+      }).not.toThrow()
+    })
+  })
+})

--- a/__tests__/spec/update-script.js
+++ b/__tests__/spec/update-script.js
@@ -157,10 +157,9 @@ describe('update.sh', () => {
     const archivePath = utils.mkReleaseArchiveSync()
     const releaseDir = path.parse(archivePath).name
 
-    // Create a git repo from the new release archive so we can see changes.
-    child_process.execSync(`unzip -q ${archivePath}`, { cwd: fixtureDir })
-    child_process.execSync(`mv ${releaseDir} ${src}`, { cwd: fixtureDir })
+    utils.mkPrototypeSync(src)
 
+    // Create a git repo from the new release archive so we can see changes.
     child_process.execFileSync('git', ['init'], { cwd: src })
     child_process.execFileSync('git', ['config', 'user.email', 'test@example.com'], { cwd: src })
     child_process.execFileSync('git', ['config', 'user.name', 'Jest Tests'], { cwd: src })

--- a/__tests__/spec/utils.js
+++ b/__tests__/spec/utils.js
@@ -60,7 +60,26 @@ function mkReleaseArchiveSync () {
   return archive
 }
 
+/**
+ * Create a test prototype from the current release archive
+ *
+ * Creates a prototype at `prototypePath`.
+ *
+ * @param {string} prototypePath
+ */
+function mkPrototypeSync (prototypePath) {
+  const archivePath = mkReleaseArchiveSync()
+  const releaseDir = path.parse(archivePath).name
+
+  const parentDir = path.dirname(prototypePath)
+  const name = path.basename(prototypePath)
+
+  child_process.execSync(`unzip -q ${archivePath}`, { cwd: parentDir })
+  child_process.execSync(`mv ${releaseDir} ${name}`, { cwd: parentDir })
+}
+
 module.exports = {
   mkdtempSync,
-  mkReleaseArchiveSync
+  mkReleaseArchiveSync,
+  mkPrototypeSync
 }

--- a/lib/build/check-files.js
+++ b/lib/build/check-files.js
@@ -1,0 +1,22 @@
+const fs = require('fs')
+const path = require('path')
+
+const repoDir = path.resolve(__dirname, '..', '..')
+
+function checkFiles () {
+  // Warn if node_modules folder doesn't exist
+  const nodeModulesExists = fs.existsSync(path.join(repoDir, '/node_modules'))
+  if (!nodeModulesExists) {
+    console.error('ERROR: Node module folder missing. Try running `npm install`')
+    process.exit(1)
+  }
+
+  // Create template .env file if it doesn't exist
+  const envExists = fs.existsSync(path.join(repoDir, '/.env'))
+  if (!envExists) {
+    fs.createReadStream(path.join(repoDir, '/lib/template.env'))
+      .pipe(fs.createWriteStream(path.join(repoDir, '/.env')))
+  }
+}
+
+checkFiles()

--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
   "scripts": {
     "start": "npm run dev",
     "build": "node lib/build/generate-assets",
+    "predev": "node lib/build/check-files",
     "dev": "node start",
     "serve": "node listen-on-port",
     "lint": "standard",

--- a/start.js
+++ b/start.js
@@ -29,22 +29,6 @@ async function collectDataUsage () {
   }
 }
 
-// Warn if node_modules folder doesn't exist
-function checkFiles () {
-  const nodeModulesExists = fs.existsSync(path.join(__dirname, '/node_modules'))
-  if (!nodeModulesExists) {
-    console.error('ERROR: Node module folder missing. Try running `npm install`')
-    process.exit(0)
-  }
-
-  // Create template .env file if it doesn't exist
-  const envExists = fs.existsSync(path.join(__dirname, '/.env'))
-  if (!envExists) {
-    fs.createReadStream(path.join(__dirname, '/lib/template.env'))
-      .pipe(fs.createWriteStream(path.join(__dirname, '/.env')))
-  }
-}
-
 function createSessionDataDefaults () {
 // Create template session data defaults file if it doesn't exist
   const dataDirectory = path.join(__dirname, '/app/data')
@@ -63,7 +47,6 @@ function createSessionDataDefaults () {
 }
 
 (async () => {
-  checkFiles()
   createSessionDataDefaults()
   await collectDataUsage()
   await buildWatchAndServe()

